### PR TITLE
fix: extract primitive data type from action/function import response

### PR DESCRIPTION
**Pull request by jjtang1985**
## Context

- [x] add function e2e tests
- [x] add action e2e tests
- [x] fix the failing action/function import tests, due to bugs of deserialisation
- [x] the root cause of the failing function tests introduced in the e2e test is clear.
- [x] generate all vdms

#### fixed deserialisation issue
The bug is related in the cases, when the return value of the function import (v2+v4) and action import is a primitive data type.
- v4 function/action import return json example, where `value` is a fixed key:
```
{
    "@odata.context": "$metadata#Edm.Int32",
    "value": 777
}
```

- v2 function import return json, where the key is using dynamical function name defined in the metadata:
```
{
    "functionName": 123
}
```

#### root cause of the failing e2e tests (disabled)
The v4 function import uses the same implementation as v2, while they are totally different. See doc [here](http://docs.oasis-open.org/odata/odata/v4.0/os/part1-protocol/odata-v4.0-os-part1-protocol.html#_Toc372793741).
v4 function Url examples:
- `http://host/service/MyShoppingCart()`
- `http://host/service/EmployeesByManager(ManagerID=3)`

v2 example:
- `https://services.odata.org/OData/OData.svc/GetProductsByRating?rating=5`

The fix (re-implemetaion) of the v4 function will be handled in a separate PR.

## Definition of Done

Please consider all items and remove only if not applicable.

- [x] Tests created/adjusted for your changes.
- [x] Release notes updated.
  * Provide sufficient context so that each entry can be understood on its own.
  * Be specific about names of functions, classes, modules, etc.
  * Describe when or where this is relevant
  * Use indicative and present tense. For example, write "Provide function `name` that does X in order to Y" over "Now X can be done by calling a new function".
- [x] PR title adheres to [conventional commit guidelines](https://www.conventionalcommits.org) (please note that only `fix:` and `feat:` will end up in the release notes)
- [ ] If applicable: Properly documented (JSDoc of public API)
- [ ] If applicable: Check if `node run doc` still works.

### CODE REVIEW

**Comment by jjtang1985** [comment]
Since the vdm updates are related to v2/v4 unit tests, generator tests and e2e tests, I have to generate all the vdms.

File: packages/generator/src/edmx-to-vdm/common/action-function-return-types.ts [hunk 1]
```diff
@@ -149,3 +162,5 @@ function getComplexReturnType(
     isCollection
   };
 }
+
+export type extractResponse = (string) => string;
```
Comment (FrankEssenberger) [inline packages/generator/src/edmx-to-vdm/common/action-function-return-types.ts:166]: Minor: I think we use capital letters for types.

File: packages/generator/src/edmx-to-vdm/v2/function-import.ts [hunk 2]
```diff
@@ -6,6 +6,9 @@ import { parseFunctionImports } from '../../edmx-parser/v2';
 import { ServiceMetadata } from '../../edmx-parser/edmx-file-reader';
 import { parseFunctionImportReturnTypes } from '../common/action-function-return-types';
 
+const extractResponse = (functionName: string) => (response: string) =>
+  `${response}.${functionName}`;
```
Comment (FrankEssenberger) [inline packages/generator/src/edmx-to-vdm/v2/function-import.ts:10]: Where does the name extractResponse come from? It look like this methods adjustes the value alias `val` with some prefix? Could we find a more descriptive name?

**Review by FrankEssenberger — APPROVED** [review]
LGTM only some minor questions and comments.

**Pull request merged**